src/core/pointcloud/qgseptdecoder.h

```diff
@@ -24,19 +24,173 @@
 #include "qgspointcloudblock.h"
 #include "qgspointcloudattribute.h"
 
+#include "laz-perf/io.hpp"
+#include "laz-perf/common/common.hpp"
+
 ///@cond PRIVATE
 #define SIP_NO_FILE
 
 #include <QString>
 
 namespace QgsEptDecoder
 {
+  struct ExtraBytesAttributeDetails
+  {
+    QString attribute;
+    QgsPointCloudAttribute::DataType type;
+    int size;
+    int offset;
+  };
+
   QgsPointCloudBlock *decompressBinary( const QString &filename, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, const QgsVector3D &scale, const QgsVector3D &offset );
   QgsPointCloudBlock *decompressBinary( const QByteArray &data, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, const QgsVector3D &scale, const QgsVector3D &offset );
   QgsPointCloudBlock *decompressZStandard( const QString &filename, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, const QgsVector3D &scale, const QgsVector3D &offset );
   QgsPointCloudBlock *decompressZStandard( const QByteArray &data, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, const QgsVector3D &scale, const QgsVector3D &offset );
   QgsPointCloudBlock *decompressLaz( const QString &filename, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, const QgsVector3D &scale, const QgsVector3D &offset );
   QgsPointCloudBlock *decompressLaz( const QByteArray &data, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes, const QgsVector3D &scale, const QgsVector3D &offset );
+
+  //! Returns the list of extrabytes attributes with their type, size and offsets represented in the LAS file
+  template<typename FileType>
+  QVector<QgsEptDecoder::ExtraBytesAttributeDetails> readExtraByteAttributes( FileType &file )
```

Comment:
Do we need this as a separate function? There's already a `laszip::io::reader::basic_file<FileType>` which could be used when this function is called.

If yes, can the code be moved to the cpp file instead?

Reply:
I wanted to do this in a seperate function to test it properly and make it modular.
As for reusing laszip::io::reader::basic_file<FileType>, I prefer not to use the internals of the laszip library since it doesn't give us the freedom of management that std file streams does.

Reply:
That function is a template and I can't put its definition inside the .cpp file.